tests/cecil-tests/GetterExceptionTest.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using Mono.Cecil;
+using System.Collections.Generic;
+using NUnit.Framework;
+using System.Linq;
+using Mono.Cecil.Cil;
+using Xamarin.Utils;
+using Xamarin.Tests;
+#nullable enable
+
+namespace Cecil.Tests {
+	[TestFixture]
+	public class GetterExceptionTest {
+		readonly HashSet<string> exceptionsToSkip = new ()
+		{
+			"AudioQueueException",
+			"ModelNotImplementedException",
+			"You_Should_Not_Call_base_In_This_Method",
+		};
+
+		bool IsMemberObsolete (ICustomAttributeProvider member)
+			 => member?.CustomAttributes?.Any ((m) =>
+					m.AttributeType.Name == "ObsoleteAttribute" ||
+					m.AttributeType.Name == "AdviceAttribute" ||
+					m.AttributeType.Name == "ObsoletedOSPlatformAttribute") == true;
+
+		bool VerifyIfGetterThrowsException (MethodDefinition methodDefinition, out string exceptionMessage)
+		{
+			exceptionMessage = string.Empty;
+			if (!methodDefinition.HasBody)
+				return false;
+
+			foreach (Instruction? inst in methodDefinition.Body.Instructions) {
+				if (inst?.OpCode == OpCodes.Newobj && inst?.Operand is MemberReference reference) {
+					TypeReference? baseType = (reference.DeclaringType as TypeDefinition)?.BaseType;
+					if (baseType is not null && baseType.Is ("System", "Exception") &&
```

Comment:
This doesn't catch exceptions that aren't direct subclasses of System.Exception (for instance System.ArgumentNullException, whose base type is System.ArgumentException, and not System.Exception).

You'd need to implement a recursive function like this (except checking for System.Exception instead of Foundation.NSObject): https://github.com/xamarin/xamarin-macios/blob/fc487c2f5904dd1e0c6da8b7376a75234c902f8a/tests/cecil-tests/ConstructorTest.cs#L191-L200